rust-runtime/aws-smithy-async/src/rt/sleep.rs

```diff
@@ -39,13 +39,22 @@ where
     }
 }
 
-#[cfg(feature = "rt-tokio")]
+#[cfg(all(
+    feature = "rt-tokio",
+    not(all(target_family = "wasm", target_os = "wasi"))
+))]
 /// Returns a default sleep implementation based on the features enabled
 pub fn default_async_sleep() -> Option<Arc<dyn AsyncSleep>> {
     Some(sleep_tokio())
 }
 
-#[cfg(not(feature = "rt-tokio"))]
+#[cfg(all(target_family = "wasm", target_os = "wasi"))]
+/// Returns a default sleep implementation based on the features enabled
+pub fn default_async_sleep() -> Option<Arc<dyn AsyncSleep>> {
+    Some(sleep_wasi())
+}
```

Comment:
this should be isolated to the WASM crate

Reply:
the sleep has been removed completely. it will probably only be necessary for the browser at this point